### PR TITLE
Change the enable/disable method

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes, and have included unit tests where possible.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish and how does it accomplish it?:**

Change the method we use to disable/enable blocking. So far, we commented the config lines
```
addn-hosts=/etc/pihole/gravity.list
addn-hosts=/etc/pihole/black.list
```
and then restarting `pihole-FTL`. This is very inefficient and slow, as, on shutdown of `pihole-FTL`, we update the database and then on restart, import all queries again from the database.

This PR changes this. Instead of changing the configuration to not load the files, we replace them by empty files which are then reread on `reload`ing `pihole-FTL`. The old files are kept as backup and restored on `pihole enable`. This is a much cleaner variant of handling the blocking status and also *much* faster than always restarting the daemon. It solves a number of delay issues recently reported with the Enable/Disable functionality of Pi-hole.

Pi-hole `gravity` is modified to update the backup files when blocking is disabled.

The current blocking status is tracked by setting the new variable `BLOCKED_ENABLED` in `setupVars.conf`.

### CODE REVIEW

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/disable-enable-feature-in-pihole-admin-web-console-hanging/11784/18

**Comment by diginc** [comment]
would `.disabled` be a more meaningful file extension than `.bck`

Does this impact the web interfaces updating of blacklist when the blacklist is in disabled mode?

**Comment by diginc** [comment]
> I don't see how symlinks can be more efficient here.

The web interface impact got me thinking about keeping the file location consistent.  I'm not sure how many locations need updated to reference both the enabled/disabled versions and lookup the setupVar, but I was trying to think of ways to abstract that away by keeping the file location static.  

> Thanks for pointing this out. I addressed this in f794018

If this is the only other spot needing updated to reflect the new disabled file location, then my thought maybe overkill.  

What I had in my head was very similar to what nginx/apache do for webserver config folders being disabled and enabled: 

```
/etc/pihole/enabled-blacklists/gravity.list (symlink) -> /etc/pihole/available-blacklists/gravity.list (always present)
/etc/pihole/enabled-blacklists/black.list (symlink) -> /etc/pihole/available-blacklists/black.list (always present)
```

Disabling would just run an `unlink /etc/pihole/enabled-blacklists/gravity.list /etc/pihole/enabled-blacklists/black.list`.  Enabling would re create the symlink.  Not necessarily better, just follows a familiar pattern to keep consistent file locations.

**Comment by DL6ER** [comment]
The web interface does it via a `pihole -b ...` call in the background so it is covered by my most recent change.

screenshot at 2018-08-13 18-48-43

**Comment by PromoFaux** [comment]
Before:
lag

After:
nolag

File: pihole [hunk 1]
```diff
@@ -187,19 +192,25 @@ Time:
...
     echo -e "  ${INFO} Enabling blocking"
     local str="Pi-hole Enabled"
 
-    sed -i 's/^#addn-hosts/addn-hosts/' /etc/dnsmasq.d/01-pihole.conf
-    if [[ -e "/etc/pihole/wildcard.list" ]]; then
-      mv "/etc/pihole/wildcard.list" "$wildcardlist"
+    if [[ -e "${gravitylist}.bck" ]]; then
+      mv "${gravitylist}.bck" "${gravitylist}"
+    fi
+    if [[ -e "${blacklist}" ]]; then
```
Comment (Mcat12) [inline pihole]: Shouldn't this check for `"${blacklist}.bck"`?

File: pihole [hunk 2]
```diff
@@ -257,16 +268,17 @@ statusFunc() {
     return 0
   fi
 
-  # Determine if Pi-hole's addn-hosts configs are commented out
-  addnConfigs=$(grep -i "addn-hosts=/" /etc/dnsmasq.d/01-pihole.conf)
+  # Determine if Pi-hole's blocking is enabled
+
+  addnConfigs=$?
```
Comment (Mcat12) [inline pihole]: This line is no longer needed

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/backup-dns-while-doing-tasks/14127/10

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/pi-hole-v4-1-is-now-available/14990/1

**Pull request merged**